src/stdlib/contains_all.rs

```diff
@@ -0,0 +1,152 @@
+use crate::compiler::prelude::*;
+use crate::stdlib::string_utils::convert_to_string;
+
+fn contains_all(value: Value, substrings: Value, case_sensitive: bool) -> Resolved {
+    let value_string = convert_to_string(value, case_sensitive)?;
+    let substring_values = substrings.try_array()?;
+
+    for substring_value in substring_values {
+        let substring = convert_to_string(substring_value, case_sensitive)?;
+        if !value_string.contains(&substring) {
+            return Ok(false.into());
+        }
+    }
+    Ok(true.into())
+}
+
+#[derive(Clone, Copy, Debug)]
+pub struct ContainsAll;
+
+impl Function for ContainsAll {
+    fn identifier(&self) -> &'static str {
+        "contains_all"
+    }
+
+    fn parameters(&self) -> &'static [Parameter] {
+        &[
+            Parameter {
+                keyword: "value",
+                kind: kind::BYTES,
+                required: true,
+            },
+            Parameter {
+                keyword: "substrings",
+                kind: kind::ARRAY,
+                required: true,
+            },
+            Parameter {
+                keyword: "case_sensitive",
+                kind: kind::BOOLEAN,
+                required: false,
+            },
+        ]
+    }
+
+    fn compile(
+        &self,
+        _state: &state::TypeState,
+        _ctx: &mut FunctionCompileContext,
+        arguments: ArgumentList,
+    ) -> Compiled {
+        let value = arguments.required("value");
+        let substrings = arguments.required("substrings");
+        let case_sensitive = arguments.optional("case_sensitive").unwrap_or(expr!(true));
```

Comment:
I generally prefer making the expression optional in `ContainsAllFn` over creating a new `true` expression, but it's not a big deal here. In general it's faster to have an optional so you don't have to evaluate an expression when we are using the "default" value.